### PR TITLE
Add workflow for checking external links in docs

### PR DESCRIPTION
**Pull request by CodyCBakerPhD**
This workflow has been around for a while, but it hasn't made its way here yet.

Now it has, though.

### CODE REVIEW

**Comment by CodyCBakerPhD** [comment]
Looks like a few aren't in fact working: https://github.com/catalystneuro/neuroconv/runs/8040199100?check_suite_focus=true

@h-mayorquin or @weiglszonja could one of you take a look and make updates when you get some time?

**Comment by h-mayorquin** [comment]
The two links that the functionality indicate as broken are false positives. They do work. I am not sure why this is failing. Some problem with links with anchors (#) but looking the [source code ](https://www.sphinx-doc.org/en/master/_modules/sphinx/builders/linkcheck.html)this should be handled. No idea.

File: .github/workflows/test_external_links.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+      - uses: actions/checkout@v2
+      - run: git fetch --prune --unshallow --tags
+      - name: Setup Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.9
+      - name: Install sphinx Dependencies
+        run: pip install -r requirements-rtd.txt
+      - name: Install package for API docs
+        run: pip install -e .
```
Comment (h-mayorquin) [inline .github/workflows/test_external_links.yml:19]:
```suggestion
        run: pip install -e .[docs]
```
This should remove the need of installing `requirements.rtd`

**Comment by CodyCBakerPhD** [comment]
@h-mayorquin any luck figuring out this one?

**Comment by h-mayorquin** [comment]
> @h-mayorquin any luck figuring out this one?

No luck so far. I had the hope that other would help.

**Comment by h-mayorquin** [comment]
@CodyCBakerPhD last thing of today, this should get us going. Apparently checking the anchors in some sites is not trivial. I still need to read more about it but this should make it work at least at the link level.

**Comment by CodyCBakerPhD** [comment]
Thanks - relaxing that can be a longer-term thing, I did just want the main workflow here for our own sanity.

**Review by h-mayorquin — APPROVED** [review]

**Pull request merged**